src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -257,9 +267,9 @@ fn search(
         assert!(lower_limit_sectors + lower_meta * 2u64 <= total_space);
         Ok((lower_limit_sectors, lower_meta))
     } else if datablocks_to_sectors(half) + half_meta * 2u64 > total_space {
```

Comment:
suggest extra condition here to allow for the unlikely possibility that ```current_data_size``` > than optimum data size.

```
if datablocks_to_sectors(half) + half_meta * 2u64 > total_space && half > current_data_size { ...
```